### PR TITLE
Updated RegExp matcher to return the matched string

### PR DESCRIPTION
**Pull request by martin770**
I like the functionality provided by gulp-contains when searching file content with a regular expression. It is somewhat useful to tell me that the regular expression was found, but this patch set provides even more value by returning the string that was found to match the regular expression.

### CODE REVIEW

**Comment by martin770** [comment]
Added a test for the functionality

**Comment by callumacrae** [comment]
Sorry for the delay in getting to this! Just one comment.

File: index.js [hunk 1]
```diff
@@ -49,7 +49,8 @@ function stringMatches(str, search) {
 		return (str.indexOf(search) !== -1) ? search : false;
 	}
 	if (search instanceof RegExp) {
-		return (str.match(search)) ? search : false;
+		// if the RegExp is found, return the matched string
+		return (search.test(str)) ? search.exec(str)[0] : false;
```
Comment (callumacrae) [inline index.js:53]: Instead of executing the regex twice, can you store `search.exec(str)` in a variable and test for null?

**Comment by callumacrae** [comment]
@martin770: Any update on my comment here?

**Comment by martin770** [comment]
Sorry, I'm really not good at github yet. I've been trying to get back to this patch set locally so that I can make the change and push it up, but my environment is not great and I haven't had enough time to figure it out. I'm ok with you modifying the code to merge it.

**Comment by callumacrae** [comment]
I can do that. Alternatively, I can work you through the process and we can do it together :)

Your call!

**Comment by martin770** [comment]
Please make the changes if you really don't mind. Some day I will get better at github... :)
  Thanks!

**Comment by marknadig** [comment]
would love to see this merged

**Pull request closed without merging**